Makefile

```diff
@@ -448,25 +448,27 @@ staging: doc/CHANGES
 	# Delete the tmp files
 	cd ${TMPDIR} && /bin/rm -rf daikon daikon.dist old_tar.txt new_tar.txt
 
-# Copy the files in the staging area to the website.  This will copy
-# all of the files in staging, but will not delete any files in the website
-# that are not in staging.
-staging-to-www: $(STAGING_DIR)
-#copy the files
+# Copy the files in the staging area to the website.
+# We do this by deleting the previous release directory and
+# renaming the staging directory to be the release directory.
+staging-to-www: $(STAGING_DIR) save-current-release
+	@echo "staging-to-www is disabled while we test new layout."
+	@exit 1
+	-chmod u+w $(WWW_PARENT)
 	-chmod -R u+w $(WWW_DIR)
-# # remove previous release archive files
-# 	\rm -f /cse/web/research/plse/daikon/download/daikon-*
-# # don't trash existing history directory
-# #	(cd $(STAGING_DIR) && tar cf - --exclude=history .) | (cd $(WWW_DIR) && tar xfBp -)
-# Remove previous files, except history and staging-daikon
-	(cd $(STAGING_DIR) && \ls | grep -v history | grep -v staging-daikon | xargs \rm -rf
-# Copy directory
-	rsync -Cavz --quiet $(STAGING_DIR) $(WWW_DIR)
-#	-chmod -R u-w $(WWW_DIR)
+	-chmod -R u+w $(STAGING_DIR)
+# Move 'permanent' directories that were not included in staging
```

Comment:
I suggest that these directories be copied (be sure to use -p to preserve timestamps) to the staging website, as part of "make staging".  That way the staging website is more complete, which is good to avoid confusion, to make links work, etc.  It's also shorter in the Makefile than the current scheme, since this target will get shorter than the "make staging" target will get longer.